catalog/core/catalog-core-solrcommands/src/test/java/org/codice/ddf/commands/solr/TestBackupCommand.java

```diff
@@ -142,6 +199,296 @@ public void testSystemPropertiesNotSet() throws Exception {
         backupCommand.doExecute();
     }
 
+    @Test
+    public void testPerformSolrCloudSynchronousBackup() throws Exception {
+
+        // Set the solr client type system property so that the
+        // BackupCommand knows that it needs to backup solr cloud.
+        setupSolrClientType(SOLR_CLOUD_CLIENT_TYPE);
+
+        // Setup BackupCommand
+        BackupCommand backupCommand = new BackupCommand() {
+            @Override
+            SolrClient getSolrClient() {
+                return miniSolrCloud.getSolrClient();
+            }
+
+            // We get the solr client from the MiniSolrCloudCluster, so we don't
+            // want to shut it down after each test. We don't create a MiniSolrCloudCluster
+            // for each test to reduce the time it takes to run the tests.
```

Comment:
I agree with Shaun. It may add complexity, but would make this more readable if there was an inner class in this test class that overrides these methods.